internal/eval/countminsketch.go

```diff
@@ -0,0 +1,538 @@
+	originalCopy := c.DeepCopy()
+
+	// resets the destination sketch
+	for row := uint64(0); row < c.opts.depth; row++ {
+		for col := uint64(0); col < c.opts.width; col++ {
+			c.matrix[row][col] = 0
+		}
+	}
+
+	// for every row and column, take the weighted sum of the source sketches.
+	for row := uint64(0); row < c.opts.depth; row++ {
+		for col := uint64(0); col < c.opts.width; col++ {
+			for i, cms := range sources {
+				if keys[i] == originalKey {
+					// use the deep copy of the destination
+					c.matrix[row][col] += weights[i] * originalCopy.matrix[row][col]
+				} else {
+					c.matrix[row][col] += weights[i] * cms.matrix[row][col]
+				}
+			}
+		}
+	}
+
+	// update the count attribute
+	c.count = 0
+	for i, cms := range sources {
+		if keys[i] == originalKey {
+			c.count += weights[i] * originalCopy.count
+		} else {
+			c.count += weights[i] * cms.count
+		}
+	}
+}
+
+// evalCMSMerge is used to merge multiple sketches into one. The final sketch
+// contains the weighted sum of the values in each of the source sketches. If
+// weights are not provided, default is 1.
+func evalCMSMerge(args []string, store *dstore.Store) *EvalResponse {
+	if len(args) < 3 {
+		return &EvalResponse{
+			Result: nil,
+			Error:  diceerrors.ErrWrongArgumentCount("cms.merge"),
+		}
+	}
+
+	destination, err := getCountMinSketch(args[0], store)
+	if err != nil {
+		return &EvalResponse{
+			Result: nil,
+			Error:  diceerrors.ErrGeneral(fmt.Sprintf("%v for 'cms.merge' command", err)),
+		}
+	}
+
+	numberOfKeys, err := strconv.ParseInt(args[1], 10, 64)
+
+	if err != nil {
+		return &EvalResponse{
+			Result: nil,
+			Error:  diceerrors.ErrGeneral("cannot parse number for 'cms.merge' command"),
+		}
+	}
+
+	if numberOfKeys < 1 {
+		return &EvalResponse{
+			Result: nil,
+			Error:  diceerrors.ErrGeneral("invalid number of arguments to merge for 'cms.merge' command"),
+		}
+	}
+
+	if len(args) != int(2+numberOfKeys) && len(args) != int(3+2*numberOfKeys) {
+		return &EvalResponse{
+			Result: nil,
+			Error:  diceerrors.ErrGeneral("invalid number of arguments to merge for 'cms.merge' command"),
+		}
+	}
+	keys := args[2 : 2+numberOfKeys]
+	sources := make([]*CountMinSketch, 0, numberOfKeys)
+
+	for _, key := range keys {
+		c, err := getCountMinSketch(key, store)
+		if err != nil {
+			return &EvalResponse{
+				Result: nil,
+				Error:  diceerrors.ErrGeneral(fmt.Sprintf("%v for 'cms.merge' command", err)),
+			}
+		}
+		if c.opts.depth != destination.opts.depth || c.opts.width != destination.opts.width {
+			return &EvalResponse{
+				Result: nil,
+				Error:  diceerrors.ErrGeneral("width/depth doesn't match"),
+			}
+		}
+		sources = append(sources, c)
+	}
+
+	if len(args) == int(2+numberOfKeys) {
+		weights := slices.Repeat([]uint64{1}, int(numberOfKeys))
+		destination.mergeMatrices(sources, weights, args[0], keys)
+
+		return &EvalResponse{
+			Result: clientio.RespOK,
```

Comment:
Hi @hgupta12. Since your commands are migrated, return clientio.OK instead of clientio.RespOK.

Reply:
Yes this worked! Thanks a lot for the review.